rpc/rest/security/auth_store.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	access "github.com/ligato/cn-infra/rpc/rest/security/model/access-security"
+)
+
+// AuthenticationDB is common interface to access user database/permissions
+type AuthenticationDB interface {
+	// AddUser adds new user with name, password and permission groups. Password should be already hashed.
+	AddUser(name, passwordHash string, permissions []string) error
+	// GetUser returns user data according to name, or nil of not found
+	GetUser(name string) (*User, error)
+	// SetLoginTime writes last login time for specific user
+	SetLoginTime(name string)
+	// SetLoginTime writes last logout time for specific user
+	SetLogoutTime(name string)
+	// IsLoggedOut uses login/logout timestamps to evaluate whether the user was logged out
+	IsLoggedOut(name string) (bool, error)
```

Comment:
Why is this needed? The user does not have to logout.. we don't really keep any session information so we can't say this with certainty. Token will expire and that's the logout.

Reply:
https://medium.com/devgorilla/how-to-log-out-when-using-jwt-a8c7823e8a6

Reply:
I'm not saying anything against logout feature and also nothing about storing tokens. It's just that the logic how the logout is made now still uses some stored state, which will actually not be stored if application restarts. In the article they imply that the token should be replaced with one with very short expiration time, or even expired. Which after replacing it in the cookie will make it unable to login even if application restarts, because the token will be invalid.

Reply:
Replace a cookie with short expiration time is one thing (and you can easily get around it copying old token and putting it back after log out), but we cannot invalidate tokens obtained as plain text on the client side. 

I think we should decide what is more important - to have a strict logout, or if we want login to survive application restart.